vtl-engine/src/test/java/fr/insee/vtl/engine/utils/dag/DagTest.java

```diff
@@ -119,7 +142,30 @@ void testDagCycle() {
     assertThatThrownBy(() -> engine.eval("e := a; b := a; c := b; a := c; f := a;"))
         .isInstanceOf(VtlScriptException.class)
         .hasMessage(
-            "Cycle detected in Script. The following statements form at least one cycle: b := a ;c := b ;a := c ");
+            "Cycle detected in Script. The following statements form 1 cycle(s): [a := c; b := a; c := b;]");
```

Comment:
I think we need to be precise about the error message.

I suggest to position the error at the last assignment in the chain and use a message like: 

```
assignment creates a cycle: [a=c=b=a]
```

Let's imagine that the error get picked up by an editor and put next to the position of the exception: 

```
e := a;
b := a;
c := b;
a := c; <- assignment creates a cycle: [a=c=b=a]
f := a;
```

Reply:
Furthermore, you might want to create your own specialised exception (extending the VtlMultiScriptException) that could contain the different cycles.